spec/spec_helper_acceptance.rb

```diff
@@ -33,6 +34,61 @@
         host.install_package('centos-release-scl-rh')
         host.install_package('rh-redis5-redis')
       end
+
+      if File.exist?('spec/acceptance/hieradata')
+        hierarchy = [
+          'fqdn/%{fqdn}.yaml',
+          'os/%{os.family}/%{os.release.major}.yaml',
+          'os/%{os.family}.yaml',
+          'common.yaml',
+        ]
+        write_hiera_config_on(host, hierarchy)
+        copy_hiera_data_to(host, 'spec/acceptance/hieradata')
+      end
+
+      apply_manifest_on(
+        # /etc/foreman-certs because /etc/foreman gets purged by apt
+        host, <<-PUPPET
+          $directory = '/etc/foreman-certs'
+          $certificate = "${directory}/certificate.pem"
+          $key = "${directory}/key.pem"
+          exec { 'Create certificate directory':
+            command => "mkdir -p ${directory}",
+            path    => ['/bin', '/usr/bin'],
+            creates => $directory,
+          } ->
+          exec { 'Generate certificate':
+            command => "openssl req -nodes -x509 -newkey rsa:2048 -subj '/CN=${facts['fqdn']}' -keyout '${key}' -out '${certificate}' -days 365",
+            path    => ['/bin', '/usr/bin'],
+            creates => $certificate,
+            umask   => '0022',
+          } ->
+          file { [$key, $certificate]:
+            owner => 'root',
+            group => 'root',
+            mode  => '0640',
+          }
+
+          class { 'foreman::repo':
+            repo                => 'nightly',
+            gpgcheck            => false,
+            configure_epel_repo => false,
```

Comment:
Since we don't actually need EPEL anymore, should we just drop the code to manage it altogether in a separate PR?

Reply:
Yes, if it's expected to stay this way... :+1: